test_conformance/subgroups/subhelpers.h

```diff
@@ -27,11 +27,145 @@
 #undef min
 #undef max
 
-// Adjust these individually below if desired/needed
-#define G 2000
-#define L 200
+#define NON_UNIFORM_WG_SIZE 4
 
-#define NON_UNIFORM 4
+
+enum class SubgroupsBroadcastOp
+{
+    broadcast,
+    broadcast_first,
+    non_uniform_broadcast
+};
+
+enum class NonUniformVoteOp
+{
+    elect,
+    all,
+    any,
+    all_equal
+};
+
+enum class BallotOp
+{
+    ballot,
+    inverse_ballot,
+    ballot_bit_extract,
+    ballot_bit_count,
+    ballot_inclusive_scan,
+    ballot_exclusive_scan,
+    ballot_find_lsb,
+    ballot_find_msb,
+    eq_mask,
+    ge_mask,
+    gt_mask,
+    le_mask,
+    lt_mask,
+};
+
+enum class ShuffleOp
+{
+    shuffle,
+    shuffle_up,
+    shuffle_down,
+    shuffle_xor
+};
+
+enum class ArithmeticOp
+{
+    add_,
+    max_,
+    min_,
+    mul_,
+    and_,
+    or_,
+    xor_,
+    logical_and,
+    logical_or,
+    logical_xor
+};
+
+static const char *const operation_names(ArithmeticOp operation)
+{
+    switch (operation)
+    {
+        case ArithmeticOp::add_: return "add"; break;
+        case ArithmeticOp::max_: return "max"; break;
+        case ArithmeticOp::min_: return "min"; break;
+        case ArithmeticOp::mul_: return "mul"; break;
+        case ArithmeticOp::and_: return "and"; break;
+        case ArithmeticOp::or_: return "or"; break;
+        case ArithmeticOp::xor_: return "xor"; break;
+        case ArithmeticOp::logical_and: return "logical_and"; break;
+        case ArithmeticOp::logical_or: return "logical_or"; break;
+        case ArithmeticOp::logical_xor: return "logical_xor"; break;
+        default: log_error("Unknown operation request"); break;
```

Comment:
This case seems to be "impossible" or occur only on internal bugs. I think this ought to abort the process, but that's just my opinion - perhaps others can comment on this.